device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/Mqtt.java

```diff
@@ -154,13 +152,27 @@ void disconnect() throws TransportException
                 }
 
                 log.debug("Sent MQTT DISCONNECT packet was acknowledged");
-                this.mqttAsyncClient.close();
             }
         }
         catch (MqttException e)
         {
-            log.warn("Exception encountered while sending MQTT DISCONNECT packet", e);
-            throw PahoExceptionTranslator.convertToMqttException(e, "Unable to disconnect");
+            // If the SDK's disconnect message doesn't make it to the service, just clean up the local resources.
+            // The service will eventually figure out that the connection was lost since the keep-alive pings stop.
+            TransportException transportException = PahoExceptionTranslator.convertToMqttException(e, "Unable to disconnect");
+            log.warn("Exception encountered while sending MQTT DISCONNECT packet. Forcefully closing the connection.", transportException);
+        }
+        finally
+        {
+            try
+            {
+                this.mqttAsyncClient.close();
+            }
+            catch (MqttException ex)
+            {
+                // As per the documentation for mqttAsyncClient.close() this exception is only thrown if the client is already
+                // closed. No need to re-attempt anything here.
+                log.debug("Mqtt client was already closed, so ignoring the thrown exception", ex);
```

Comment:
I need to double check that the exception is a part of this log statement

Reply:
Yep, this does work the way I thought. Even though there is no "{}" space for it, log.debug knows to print the exception with the base message